pkg/controller/jobs/statefulset/statefulset_webhook.go

```diff
@@ -53,7 +53,7 @@ func SetupWebhook(mgr ctrl.Manager, opts ...jobframework.Option) error {
 		Complete()
 }
 
-// +kubebuilder:webhook:path=/mutate-apps-v1-statefulset,mutating=true,failurePolicy=fail,sideEffects=None,groups="apps",resources=statefulsets,verbs=create,versions=v1,name=mstatefulset.kb.io,admissionReviewVersions=v1
+// +kubebuilder:webhook:path=/mutate-apps-v1-statefulset,mutating=true,failurePolicy=fail,sideEffects=None,groups="apps",resources=statefulsets,verbs=create;update,versions=v1,name=mstatefulset.kb.io,admissionReviewVersions=v1
```

Comment:
This does not seem needed - we don't mutate on update.

Reply:
Yes, we need it because we should update `kueue.x-k8s.io/pod-group-total-count` annotation.